lib/mutgen.c

```diff
@@ -402,8 +476,13 @@ mutgen_generate(mutgen_t *self, tsk_table_collection_t *tables, int flags)
         ret = MSP_ERR_INCOMPATIBLE_MUTATION_MAP;
         goto out;
     }
+    if (flags & MSP_DISCRETE_SITES) {
+        max_num_samples = 100;
```

Comment:
Ah, this caught me out. I interpreted ``samples`` to mean sample individuals in the usual tskit sense, not the maximum number of rejection samples. Maybe different terminology here?